### PR TITLE
Explore: simplify support for multiple query editors

### LINKED ISSUE #48175: Explore: Deprecate / remove multiple ways of setting Query editors in Explore [9.0.0]
[Slack thread](https://raintank-corp.slack.com/archives/C02TXB92G6T/p1643965091137989)

as pointed out by @ivanahuckova we have multiple ways of defining query editors for explore, some of them are leftovers from when we had a manual switch between metric & logs editors.

Do we need all of them or we can remove some?

We should also document how to define different editors for explore/dashboards/alerting

### PR DESCRIPTION
**Pull request by Elfo404**
**What this PR does / why we need it**:
Deprecates `setExplore*QueryField` plugins configuration options in favor of a unified API using only `setQueryEditor` and updates CloudWatch to remove usage of those deprecated settings.

**Which issue(s) this PR fixes**:

Fixes #48175 

**Special notes for your reviewer**:
I didn't want the whole company for a review 🙏 

# Deprecation notice

`setExploreQueryField`, `setExploreMetricsQueryField` and `setExploreLogsQueryField` are now deprecated and will be removed in a future release. If you need to set a different query editor for Explore, conditionally render based on `props.app` in your regular query editor. Please refer to https://grafana.com/docs/grafana/latest/developers/plugins/add-support-for-explore-queries/ for more informations.

### CODE REVIEW

File: docs/sources/developers/plugins/add-support-for-explore-queries.md [hunk 1]
```diff
@@ -10,11 +10,11 @@ This guide assumes that you're already familiar with how to [Build a data source
 
 With Explore, users can make ad-hoc queries without the use of a dashboard. This is useful when users want to troubleshoot or to learn more about the data.
 
-Your data source already supports Explore by default, and will use the existing query editor for the data source. If you want to offer extended Explore functionality for your data source however, you can define a Explore-specific query editor.
+Your data source already supports Explore by default, and will use the existing query editor for the data source.
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/add-support-for-explore-queries.md:13]:
```suggestion
Your data source supports Explore by default and uses the existing query editor for the data source.

```

File: docs/sources/developers/plugins/add-support-for-explore-queries.md [hunk 2]
```diff
@@ -10,11 +10,11 @@ This guide assumes that you're already familiar with how to [Build a data source
...
 
-## Add a query editor for Explore
+## Add an Explore-specific query editor
 
-The query editor for Explore is similar to the query editor for the data source itself. In fact, you'll probably reuse the same components for both query editors.
+If you want to offer extended Explore functionality for your data source, you can define an Explore-specific query editor.
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/add-support-for-explore-queries.md:17]:
```suggestion
If you want to extend Explore functionality for your data source, you can define an Explore-specific query editor.
```

File: docs/sources/developers/plugins/add-support-for-explore-queries.md [hunk 3]
```diff
@@ -26,60 +26,31 @@ The query editor for Explore is similar to the query editor for the data source
...
+   type Props = QueryEditorProps<DataSource, MyQuery, MyDataSourceOptions>;
 
    export default (props: Props) => {
-     return <h2>My query editor</h2>;
+     return <h2>My Explore-specific query editor</h2>;
    };
    ```
 
-1. Configure the plugin to use the `ExploreQueryEditor`.
+1. Modify your base query editor in `QueryEditor.tsx` to render the Explore-specific one when using Explore. For example:
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/add-support-for-explore-queries.md:36]:
```suggestion
1. Modify your base query editor in `QueryEditor.tsx` to render the Explore-specific query editor. For example:
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

**Review by achatterjee-grafana — APPROVED** [review]
Requesting one more change, otherwise, docs look good.

**Review by sunker — APPROVED** [review]
LGTM

**Review by ivanahuckova — APPROVED** [review]
🤩 YES!

**Pull request merged**